### ISSUE TITLE
yargs processes --no-XXX boolean options differently 

### ISSUE THREAD

**Issue by aduh95** [yargs/yargs#1011]
Greetings,

It seems that yargs have trouble parsing aliases with a dash in it. I have a `--no-browser`  option in my program but yargs does not listen to it. You can reproduce this bug with yargs v10.0.3 by creating a `yargs.js` file:

```js
console.log(
  require("yargs")
    .boolean("n")
    .alias("n", "no-browser").argv
);
```

As expected, the `-n` CLI argument triggers yargs' option:

```bash
$ node yargs.js
{ _: [],
  help: false,
  version: false,
  n: false,
  'no-browser': false,
  noBrowser: false,
  '$0': 'yargs.js' }

$ node yargs.js -n
{ _: [],
  help: false,
  version: false,
  n: true,
  'no-browser': true,
  noBrowser: true,
  '$0': 'yargs.js' }
```

But surprisingly, the `--no-browser` flag has no effect what so ever, but instead another alias is created: `--noBrowser`.

```js
$ node yargs.js --no-browser
{ _: [],
  help: false,
  version: false,
  n: false,
  'no-browser': false,
  noBrowser: false,
  browser: false,
  '$0': 'yargs.js' }

$ node yargs.js --noBrowser
{ _: [],
  help: false,
  version: false,
  n: true,
  'no-browser': true,
  noBrowser: true,
  '$0': 'yargs.js' }
```

Is there a workaround to make the `--no-browser` option works?

**Comment by nexdrew** [comment]
Hi! I think this is probably a little hard to find in the docs, but you can disable the special treatment of the `--no-` flag prefix by putting this in your package.json:

```json
  "yargs": {
    "boolean-negation": false
  }
```

This is [briefly mentioned in yargs docs here](https://github.com/yargs/yargs/blob/f8efa0168c7cd6ad5a9f2d44a787bf80c0f300fa/docs/advanced.md#customizing-yargs-parser) and also [mentioned in the yargs-parser readme here](https://github.com/yargs/yargs-parser#boolean-negation).

Give that a shot and let us know what you find. Thanks!

**Comment by aduh95** [comment]
Hey @nexdrew, thanks for your answer, it was what I was looking for, it now works as expected!

**Comment by cjauvin** [comment]
I struggled with this for a while, before finding this thread. I actually thought it was some strange bug and was getting ready to file an issue. Once I knew it wasn't, I had to search for another while in order to find the way to customize the parser without changing my `package.json`, as suggested above. In the context of the code snippet that started the thread, that would be:

```javascript
console.log(
  require("yargs")
    .parserConfiguration({
      "boolean-negation": false,
    })
    .boolean("n")
    .alias("n", "no-browser").argv
);
```

I'm not sure if this issue is very frequent, but I would be inclined to believe that being a little bit more explicit about it in the documention might be useful to some people.
